src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -968,7 +968,17 @@ private void ImportBranch(ILOpcode opcode, BasicBlock target, BasicBlock fallthr
 
         private void ImportSwitchJump(int jmpBase, int[] jmpDelta, BasicBlock fallthrough)
         {
-            throw new NotImplementedException("switch");
+            var operand = _stack.Pop();
+
+            var @switch = LLVM.BuildSwitch(_builder, operand.LLVMValue, GetLLVMBasicBlockForBlock(fallthrough), (uint)jmpDelta.Length);
+            for (var i = 0; i < jmpDelta.Length; i++)
+            {
+                var target = _basicBlocks[_currentOffset + jmpDelta[i]];
+                LLVM.AddCase(@switch, LLVM.ConstInt(LLVM.Int32Type(), (ulong)i, false), GetLLVMBasicBlockForBlock(target));
+                ImportFallthrough(target);
```

Comment:
Why did you end up calling ImportFallthrough here? It looks like for other branching we don't need that.

Reply:
I used it to add a valid terminator at the end of the switch block